setup-test-network.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+set -e
+
+# ==============================================================================
+# SETUP LOCAL GRAPH NETWORK FOR TESTING
+# ==============================================================================
+# This script sets up a local Graph network for testing.
+#
+# NOTES:
+# - If you encounter container conflicts, run: docker compose down
+#   to stop all services before running this script again
+#
+# - To test changes to your indexer code without restarting everything:
+#   make reload
+#
+# - The script checks for existing services and skips those already running
+# ==============================================================================
+#
+# Save the starting disk usage
+START_SPACE=$(df -h --output=used /var/lib/docker | tail -1)
+START_IMAGES_SIZE=$(docker system df --format '{{.ImagesSize}}' 2>/dev/null || echo "N/A")
+START_CONTAINERS_SIZE=$(docker system df --format '{{.ContainersSize}}' 2>/dev/null || echo "N/A")
+START_VOLUMES_SIZE=$(docker system df --format '{{.VolumesSize}}' 2>/dev/null || echo "N/A")
+
+echo "============ STARTING DISK USAGE ============"
+echo "Docker directory usage: $START_SPACE"
+echo "Images size: $START_IMAGES_SIZE"
+echo "Containers size: $START_CONTAINERS_SIZE"
+echo "Volumes size: $START_VOLUMES_SIZE"
+echo "=============================================="
+
+# Your existing script starts here
+container_running() {
+    docker ps --format '{{.Names}}' | grep -q "^$1$"
+    return $?
+}
+
+if container_running "indexer-service" && container_running "tap-agent" && container_running "gateway"; then
+    echo "====================================================================================="
+    echo "All services are already running. To test changes to your indexer code, you can use:"
+    echo "  make reload                  - To rebuild and restart just indexer-service tap-agent services"
+    echo ""
+    echo "If you need to start from scratch, first stop all services with:"
+    echo "  make down"
+    echo "  docker rm -f indexer-service tap-agent gateway"
+    echo "====================================================================================="
+    exit 0
+fi
+
+cd contrib/
+ls
+pwd
+
+# Clone local-network repo if it doesn't exist
+if [ ! -d "local-network" ]; then
+    git clone https://github.com/edgeandnode/local-network.git
+    cd local-network
+    # Checkout to a specific commit that is known to work
```

Comment:
Would be awesome to know what makes it work/break from this point.

Reply:
it is actually a recent commit from master. The previous issue I found was fixed.
however, because local network is under development(i noticed many changes being merge recently), i decided to pin that commit to avoid unexpected issues in the near future.